apis/v1alpha2/nginxproxy_types.go

```diff
@@ -388,6 +389,35 @@ type KubernetesSpec struct {
 	Service *ServiceSpec `json:"service,omitempty"`
 }
 
+// Patch defines a patch to apply to a Kubernetes object.
+type Patch struct {
+	// Type is the type of patch. Defaults to StrategicMerge.
+	//
+	// +optional
+	// +kubebuilder:default:=StrategicMerge
+	Type *PatchType `json:"type,omitempty"`
+
+	// Value is the patch data as raw JSON.
+	// For StrategicMerge and Merge patches, this should be a JSON object.
+	// For JSONPatch patches, this should be a JSON array of patch operations.
+	//
+	// +kubebuilder:validation:XPreserveUnknownFields
+	Value *apiextv1.JSON `json:"value"`
+}
+
+// PatchType specifies the type of patch.
+// +kubebuilder:validation:Enum=StrategicMerge;Merge;JSONPatch
+type PatchType string
+
+const (
+	// PatchTypeStrategicMerge uses strategic merge patch.
+	PatchTypeStrategicMerge PatchType = "StrategicMerge"
```

Comment:
Does k8s define these types at all? If so, it'd be nice to alias our types to theirs, instead of using the string literal.

Reply:
Not really. The definitions @bjee19 links above are MIME types. I couldn't find any appropriate user friendly, CRD style definitions we could reference.